### PR TITLE
feat!: use gql library for GraphQL requests

### LINKED ISSUE #20: Reformat query construction
Stuff like this: https://stackoverflow.com/questions/62384215/best-way-to-construct-a-graphql-query-string-in-python

### PR DESCRIPTION
**Pull request by jsstevenson**
close #20 

Use [gql](https://github.com/graphql-python/gql) for issuing requests. Makes things a bit more ergonomic and performs client-side validation during repeated requests, meaning faster failures (also enables some smarter testing on our end down the line).

I took the liberty of making a few minor tweaks to a couple of queries/function args but otherwise this should largely be a drop-in replacement, functionality staying mostly the same.

### CODE REVIEW

File: src/dgipy/dgidb.py [hunk 1]
```diff
@@ -373,7 +484,7 @@ def __process_gene_search(results: dict) -> pd.DataFrame:
     pmids_list = []
     # genecategories_list = []
 
-    for match in results["data"]["genes"]["nodes"]:
+    for match in results["genes"]["nodes"]:
```
Comment (mcannon068nw) [inline src/dgipy/dgidb.py:487]: outside of scope of this PR, but fwiw I am 100% open to refactoring this result table building method. Pretty sure there is a better way to do this, this is just what I got down at the time

File: src/dgipy/dgidb.py [hunk 2]
```diff
@@ -122,69 +150,123 @@ def get_interactions(
...
+            """
         )
     else:
         msg = "Search type must be specified using: search='drugs' or search='genes'"
         raise Exception(msg)
 
     api_url = api_url if api_url else API_ENDPOINT_URL
-    r = requests.post(api_url, json={"query": query}, timeout=20)
+    client = _get_client(api_url)
+    result = client.execute(query, variable_values=params)
```
Comment (mcannon068nw) [inline src/dgipy/dgidb.py:263]: big fan of this, def feels better than building a giant query string and sending it with requests

File: src/dgipy/dgidb.py [hunk 3]
```diff
@@ -122,69 +150,123 @@ def get_interactions(
...
-    :param sourcedbname: filter option for specific databases of interest
-    :param pmid: filter option for specific PMIDs:
-    :param interactiontype: filter option for specific interaction types
+    :param source: filter option for specific database of interest
+    :param pmid: filter option for specific PMID
+    :param interaction_type: filter option for specific interaction types
     :param approved: filter option for approved interactions
     :param api_url: API endpoint for GraphQL request
     :return: interaction results for terms in either a dataframe or a json object
     """
```
Comment (mcannon068nw) [inline src/dgipy/dgidb.py:159]: we should make sure that theres no case sensitivity issues on the resolver side as well for parameters through graphql. iirc one resolver had an issue with that while you were out and we had to change it

**Review by mcannon068nw — APPROVED** [review]
🎺 🎺

File: src/dgipy/dgidb.py (hunk 3 above)
Comment (jsstevenson) [reply src/dgipy/dgidb.py:159]: oh I believe it

**Pull request merged**